primary/src/grpc_server/configuration.rs

```diff
@@ -2,18 +2,26 @@
 // SPDX-License-Identifier: Apache-2.0
 use config::{PrimaryAddresses, SharedCommittee};
 use crypto::traits::VerifyingKey;
+use multiaddr::Multiaddr;
 use std::collections::BTreeMap;
 use tonic::{Request, Response, Status};
-use types::{Configuration, Empty, NewEpochRequest, NewNetworkInfoRequest, PublicKeyProto};
+use types::{
+    Configuration, Empty, GetPrimaryAddressResponse, MultiAddrProto, NewEpochRequest,
+    NewNetworkInfoRequest, PublicKeyProto,
+};
 
 pub struct NarwhalConfiguration<PublicKey: VerifyingKey> {
+    primary_addr: Multiaddr,
     /// The committee
     committee: SharedCommittee<PublicKey>,
 }
 
 impl<PublicKey: VerifyingKey> NarwhalConfiguration<PublicKey> {
-    pub fn new(committee: SharedCommittee<PublicKey>) -> Self {
-        Self { committee }
+    pub fn new(primary_addr: Multiaddr, committee: SharedCommittee<PublicKey>) -> Self {
```

Comment:
Since the `primary_addr` is derived from the `comittee` can't we just derive it from them when we initialise the struct, so we don't have the need to pass it separately?

Reply:
We would need the name (public key) of the primary to be able to derive it from the committee so I could either pass the name or the primary address directly in. Maybe the name is better if we ever want to derive other information from the committee?

Reply:
@akichidis I changed this to pass in the name instead of primary address, if you had something else in mind let me know and I will change it in a future PR.